controls/SV-257889.rb

```diff
@@ -19,14 +21,31 @@
 $ sudo chmod 0740 /home/wadea/.<INIT_FILE>'
   impact 0.5
   ref 'DPMS Target Red Hat Enterprise Linux 9'
-  tag check_id: 'C-61630r925652_chk'
   tag severity: 'medium'
+  tag gtitle: 'SRG-OS-000480-GPOS-00227'
   tag gid: 'V-257889'
   tag rid: 'SV-257889r925654_rule'
   tag stig_id: 'RHEL-09-232045'
-  tag gtitle: 'SRG-OS-000480-GPOS-00227'
   tag fix_id: 'F-61554r925653_fix'
-  tag 'documentable'
   tag cci: ['CCI-000366']
   tag nist: ['CM-6 b']
+  tag 'host'
+
+  only_if('This control is Not Applicable to containers', impact: 0.0) {
+    !virtualization.system.eql?('docker')
+  }
+
+  ignore_shells = input('non_interactive_shells').join('|')
+
+  homedirs = users.where { !shell.match(ignore_shells) && (uid >= 1000 || uid.zero?) }.homes
+  ifiles = command("find #{homedirs.join(' ')} -xdev -maxdepth 1 -name '.*' -type f").stdout.split("\n")
```

Comment:
this seems like it's just trying to find all one's dotfiles in one's homedir.  nominally that's wrong because there could be dotfiles in there that don't impact the shell and are used for configuring other applications, ex. .vimrc for vim.

additionally, configuration files don't necessarily need to exist in just the top level.  for example, you could have your .bashrc at the top level, but then have it source further configuration files from another location that we wouldn't find with this `find`.

furthermore, there are even specifications such as freedesktop's [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) that have gotten increasing amounts of [support ](https://wiki.archlinux.org/title/XDG_Base_Directory )where people often set their configuration files - even for shells.

i think if we wanted to do this properly we would a) keep an eye on all the dotfiles in the homedir, b) have an input that would tell us of additional locations we would need to inspect, and c) have an input that would have a list of files that we could ignore